nestkernel/node_collection.h

```diff
@@ -617,13 +626,13 @@ inline NodeIDTriple nc_const_iterator::operator*() const
   {
     // for efficiency we check each value instead of simply checking against
     // composite_collection->end()
-    if ( composite_collection_->stop_offset_ != 0 or composite_collection_->stop_part_ != 0 )
+    if ( composite_collection_->end_offset_ != 0 or composite_collection_->end_part_ != 0 )
     {
-      if ( not( part_idx_ < composite_collection_->stop_part_
-             or ( part_idx_ == composite_collection_->stop_part_
-                  and element_idx_ < composite_collection_->stop_offset_ ) ) )
+      if ( not( part_idx_ < composite_collection_->end_part_
+             or ( part_idx_ == composite_collection_->end_part_
+                  and element_idx_ < composite_collection_->end_offset_ ) ) )
       {
-        throw KernelException( "Invalid NodeCollection iterator (composite element beyond specified stop element)" );
+        throw KernelException( "Invalid NodeCollection iterator (composite element beyond specified end element)" );
       }
     }
     else if ( part_idx_ >= composite_collection_->parts_.size()
```

Comment:
This test here seems redundant. Can't we always initialise end_offset_ and end_part_ so that the first test will do and we do not need to check them against being zero?

Reply:
Maybe. I've added it to #2203.